### ISSUE TITLE
Add "mass" calculation to energy requirement

### ISSUE THREAD

**Issue by int-ua** [mt-mods/jumpdrive#60]
Simplest way: count air/vacuum blocks and exclude them from radius cubed. To discourage moving large chunks of land into the sky.

**Comment by thomasrudin** [comment]
Nice idea, what about chest contents? A gold-chest full of iron-block stacks should weigh almost as much as the chunk you are trying to move...

> To discourage moving large chunks of land into the sky.

I personally have no issues with that, but ok :smile:

**Comment by int-ua** [comment]
> Nice idea, what about chest contents? A gold-chest full of iron-block stacks should weigh almost as much as the chunk you are trying to move...

Sounds like it would take too much CPU for some ships. Unless you add caching of every ship container "mass". Which is just smearing that CPU load over time and database.

**Comment by BobFred7** [comment]
> > To discourage moving large chunks of land into the sky.
> 
> I personally have no issues with that, but ok

Yeah, could be a low-priority enhancement that can be toggled in configuration

If it would be configured anyway

> > Nice idea, what about chest contents? A gold-chest full of iron-block stacks should weigh almost as much as the chunk you are trying to move...

> Sounds like it would take too much CPU for some ships. Unless you add caching of every ship container "mass". Which is just smearing that CPU load over time and database.

Toggle for whether or not to calculate chest mass
